generator/src/test/java/org/apache/kafka/message/MessageDataGeneratorTest.java

```diff
@@ -90,4 +90,192 @@ public void testInvalidNullDefaultForPotentiallyNonNullableArray() throws Except
                     e.getMessage().contains("not all versions of this field are nullable"));
         }
     }
+
+    @Test
+    public void testInvalidFieldName() {
```

Comment:
It would be helpful to indicate what is invalid about the name. Is it the underscore at the start?

Reply:
I added some JavaDoc